src/backend/distributed/transaction/backend_data.c

```diff
@@ -401,8 +406,16 @@ StoreAllActiveTransactions(Tuplestorestate *tupleStore, TupleDesc tupleDescripto
 			continue;
 		}
 
+		PGPROC *currentProc = &ProcGlobal->allProcs[backendIndex];
+		if (currentProc->lockGroupLeader != NULL)
```

Comment:
or should we use `IsBackgroundWorker` or something similar ?

Reply:
The name of the UDF is `citus_worker_stat_activity ` , though we are planning to change it. The idea is to be able show ALL the activity going on the cluster. So, we should be able to show the background workers.

Though, thinking on this again, I think we should also be able to show parallel queries on that view as pg_stat_activity does. We might consider adding `leader_pid` to the output, such that a higher level UDF/function can filter if needed.

In other words, I decided the close this PR for now.